### ISSUE TITLE
'__construct_node' is ambiguous error under macos

### ISSUE THREAD

**Issue by zelig** [ethereum/aleth#79]
```
json_spirit/json_spirit_value.h:14:


/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/c++/v1/map:1276:29: error: call to
      member function '__construct_node' is ambiguous
        __node_holder __h = __construct_node(__k);
                            ^~~~~~~~~~~~~~~~
```

https://groups.google.com/forum/#!topic/boost-list/mcZxc6zuYOY

a libc++ bug with c++11?

http://llvm.org/bugs/show_bug.cgi?id=16549

**Comment by simondlr** [comment]
Hey @zelig. Did you eventually get it to work? Getting a similar bug when installing AlethZero. 

Apple LLVM version 5.0 (clang-500.2.79) (based on LLVM 3.3svn)
Target: x86_64-apple-darwin13.1.0
Thread model: posix

Possibly some conflict somewhere?

**Comment by simondlr** [comment]
Okay. For future reference for those who find this bug. I fixed it.

This is indeed a bug with libc++. You have to upgrade your Xcode (so that libc++ also updates). The reason is that the bug was fixed > 3.3svn of libc++. So any Xcode < 5.1 will throw this error. I had 5.0.2. From 5.1 it uses base of 3.4svn (which has the fix).

https://gist.github.com/yamaya/2924292

The Ethereum wiki is thus wrong. Someone tested on 5.0.2 and it worked, but it doesn't. https://github.com/ethereum/cpp-ethereum/wiki/Building-on-MacOS
